lib/miq_automation_engine/engine/miq_ae_service.rb

```diff
@@ -318,6 +318,40 @@ def __find_instance_from_path(path)
       aec.ae_instances.detect { |i| instance.casecmp(i.name) == 0 }
     end
 
+    def with_acquired_lock(lock_name, max_retry_time=60, min_retries=10)
+      give_up_at = Time.now + max_retry_time
+      retry_interval = max_retry_time / min_retries
+
+      while Time.now < give_up_at do
+        if MiqConcurrency::PGMutex.try_advisory_lock(lock_name)
+          begin
+            yield
+          ensure
+            MiqConcurrency::PGMutex.release_advisory_lock(lock_name)
+          end
+          return # Success
+        end
+        sleep(rand(retry_interval))
```

Comment:
@dtrieu80 
Wont randomizing the retry_interval mean that you could wait less than the max_retry_time.
Is there a reason for randomizing instead of using the retry_interval